share/disambiguation.py

```diff
@@ -144,6 +144,32 @@ def _instance_for_node(self, node):
         logger.error('Could not disambiguate %s. Too many results found from %s %s', node.model, all_query, queries)
         raise NotImplementedError('Multiple {0}s found'.format(node.model))
 
+    def _instance_for_subject(self, node):
+        # Subject disambiguation is a bit weird: Match (uri) OR (name AND taxonomy)
+        if 'uri' in node.attrs:
+            try:
+                return node.model.objects.get(uri=node.attrs['uri'])
+            except node.model.DoesNotExist:
+                pass
+
+        name = node.attrs.get('name')
+        if node.related('central_synonym') is None:
+            # Central taxonomy
+            if name:
+                try:
+                    return node.model.objects.get(name=name, central_synonym__isnull=True)
+                except node.model.DoesNotExist:
+                    logger.debug('No %s found with name "%s" in the central taxonomy', node.model, name)
+        else:
+            # Custom taxonomy
+            source = node.graph.source
+            if source and name:
+                try:
+                    return node.model.objects.get(name=name, taxonomy__name=source.long_title)
+                except node.model.DoesNotExist:
+                    logger.debug('No %s found with name "%s" in taxonomy "%s"', node.model, name, source.long_title)
```

Comment:
Similar to above. If it's, say, EngrXiv updating their custom taxonomy, nothing is wrong.